jax/numpy/lax_numpy.py

```diff
@@ -1399,6 +1401,9 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
 
   a_dtype = _dtype(a)
   if dtype:
+    if dtype not in complex_dtypes and a_dtype in complex_dtypes:
+      raise ValueError("dtype must be complex when calling var on a "
```

Comment:
Add a comment explaining the reasoning here, given NumPy doesn't error here?

Reply:
Sure, added a sentence explaining why we're raising an error.

Reply:
The issue with this error is that it isn't actionable for the user.

Perhaps change the error to something like:

jax.numpy.var does not yet support real dtype parameters when computing the variance of an array of complex values. The semantics of numpy.var seem unclear in this case. Please comment on https://github.com/google/jax/issues/2283 if this behavior is important to you.

Reply:
Changed the error message as per your suggestion.